### ISSUE TITLE
strToVersionInfo function does not reference last element of version->mExtra array correctly

### ISSUE THREAD

**Issue by StanislavBelotserkovskiy** [OpenMAMA/OpenMAMA#347]
# strToVersionInfo function does not reference last element of version->mExtra array correctly
## 
last element of version->mExtra array is referenced incorrectly - one char passed the array boundaries - when setting it to NULL 

## Areas Affected
- [ X] MAMAC
- [ X] MAMACPP
- [ ] MAMADOTNET
- [ ] MAMAJNI
- [ ] MAMDA
- [ ] MAMDACPP
- [ ] MAMDADOTNET
- [ ] MAMDAJNI
- [ ] Visual Studio
- [ ] SCons
- [ ] Unit Tests
- [ ] Examples

## Details

This line of code in strToVersionInfo function will cause random memory corruption because it updates a memory location outside of the array boundaries:

version->mExtra[VERSION_INFO_EXTRA_MAX] = '\0';

test case getVerInfoFromString in common\c_cpp\src\gunittest\c\utiltest.cpp fails because of this issue

**Comment by fquinner** [comment]
Thanks @StanislavBelotserkovskiy, looks like @dmagOM already took care of this one and it's been approved and merged into next: #310

**Comment by StanislavBelotserkovskiy** [comment]
@fquinner Should we be comparing our proposed changes with next branch or master branch when preparing a pull request?  One reason I did not realize that this issue was resolved already was that the fix was not present on master branch.

**Comment by fquinner** [comment]
Yeah next is the target branch for PRs. We follow Git Flow with `next` as our development branch: https://openmama.github.io/openmama_submission_process.html
